hs_core/hydroshare/utils.py

```diff
@@ -682,6 +684,60 @@ def validate_resource_file_count(resource_cls, files, resource=None):
                 raise ResourceFileValidationException(err_msg)
 
 
+def convert_file_size_to_unit(size, unit):
+    """
+    Convert file size to unit for quota comparison
+    :param size: in byte unit
+    :param unit: should be one of the four: 'KB', 'MB', 'GB', or 'TB'
+    :return: the size converted to the pass-in unit
+    """
+    unit = unit.lower()
+    kbsize = size / 1000.0
+    if unit == 'kb':
+        return kbsize
+    mbsize = kbsize / 1000.0
+    if unit == 'mb':
+        return mbsize
+    gbsize = mbsize / 1000.0
+    if unit == 'gb':
+        return gbsize
+    tbsize = gbsize / 1000.0
+    if unit == 'tb':
+        return tbsize
+
+    raise ValidationError('Pass-in unit for file size conversion must be one of KB, MB, GB, or TB')
+
+
+def validate_user_quota(user, size):
+    """
+    validate to make sure the user is not over quota with the newly added size
+    :param user: the user to be validated
+    :param size: the newly added file size to add on top of the user's used quota to be validated.
+                 size input parameter should be in byte unit
+    :return: raise exception for the over quota case
+    """
+    if user:
```

Comment:
Yes, since it is called with this user input parameter directly coming from get_quota_holder() call which return None if the resource has no quota holder. So this is needed at least for initial migration. Although it may not be needed in the future when we can guarantee each resource has a quota holder correctly set up, it won't hurt to have this check as a safe guard.